zkevm-circuits/src/copy_circuit.rs

```diff
@@ -301,25 +322,9 @@ impl<F: Field> SubCircuitConfig<F> for CopyCircuitConfig<F> {
                 meta.query_advice(is_pad, Rotation::next()),
             );
 
-            cb.gate(meta.query_selector(q_step))
-        });
-
-        meta.create_gate("verify_step (q_step == 0)", |meta| {
-            let mut cb = BaseConstraintBuilder::default();
-
-            cb.require_equal(
-                "rows[2].value == rows[0].value * r + rows[1].value",
-                meta.query_advice(value, Rotation(2)),
-                meta.query_advice(value, Rotation::cur()) * challenges.keccak_input()
-                    + meta.query_advice(value, Rotation::next()),
-            );
-
             cb.gate(and::expr([
                 meta.query_fixed(q_enable, Rotation::cur()),
-                not::expr(meta.query_selector(q_step)),
-                not::expr(meta.query_advice(is_last, Rotation::cur())),
-                tag.value_equals(CopyDataType::RlcAcc, Rotation::cur())(meta),
-                not::expr(meta.query_advice(is_pad, Rotation::cur())),
+                meta.query_selector(q_step),
```

Comment:
We can remove line 326 since `q_enable = 1` when `q_step = 1`.

Reply:
Sorry, I didn't see any place constraint `q_enable = 1` when `q_step = 1`. Could you please show me where?

Reply:
Since both `q_enable` and `q_step` are fixed column, and from the design (or assignment) `q_enable` is for every row (except blinding factors) and `q_step` is for every 2-row (except blinding factors).

Reply:
fixed in [d591f47](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1419/commits/d591f47ab9558031eacf11e4e04b148685db7654)

Reply:
>  from the design (or assignment) q_enable is for every row (except blinding factors) and q_step is for every 2-row (except blinding factors).

I could understand from the design we'll assign `q_enable` for every row but I thought we can't trust assignment part. Which means malicious provers could set `q_enable = 0` when `q_step = 1` and the constraints here will be bypassed.

Reply:
But for `q_enable` and `q_step` they are both preprocess polynomial, so I think we can trust the assignment, right?

Reply:
yup, make sense, you're right! thanks.